papyri/signature.py

```diff
@@ -0,0 +1,70 @@
+import inspect
+
+
+class Signature:
+    """A wrapper around inspect utilities."""
+
+    def __init__(self, target_item):
+        """
+        :param target_item: A callable
+        """
+        self.target_item = target_item
+        self._sig = inspect.signature(target_item)
+
+    @property
+    def parameters(self):
+        return self._sig.parameters
+
+    @property
+    def is_async(self):
+        return inspect.iscoroutinefunction(self.target_item)
```

Comment:
Does `is_async` as a name make sens ? You yourself as below "is_async_generator".

Reply:
> Ah, yes that's probably not. What do you think about `is_async_function` (to differentiate with `is_async_generator`)?

I'm not sure this is 100% necessary, keep in mind we are mostly interested in the json that will be stored, and here we are discussing helper method. 

Have a look at in progress https://github.com/aktech/papyri/compare/griffe-signature...Carreau:papyri:griffe-signature?expand=1.

Reply:
I see, what you mean. That looks like a good starting point. Would you like to your branch here, so that I can do some cleanups and merge?

Reply:
Sure, I just rebased on yours and pushed here.